shifthelper/checks.py

```diff
@@ -50,35 +50,223 @@ def inner_check(self):
-            return "relative camera temperature > 15°C"
+            return 'relative camera temperature > 15°C'
+
+
+class BiasNotOperatingDuringDataRun(FactIntervalCheck):
+    def inner_check(self):
+        if (is_bias_not_operating()
+                and is_data_taking()
+                and is_data_run()):
+            return 'Bias not operating during data run'
+
+
+class BiasChannelsInOverCurrent(FactIntervalCheck):
+    def inner_check(self):
+        bias_state = sfc.status()['Bias_control']
+        if bias_state == 'OverCurrent':
+            return 'Bias Channels in Over Current'
+
+
+class BiasVoltageNotAtReference(FactIntervalCheck):
+    def inner_check(self):
+        bias_state = sfc.status()['Bias_control']
+        if bias_state == 'NotReferenced':
+            return 'Bias Voltage not at reference.'
+
+
+class ContainerTooWarm(FactIntervalCheck):
+    def inner_check(self):
+        if sfc.container_temperature()['Current_temperature_in_C'] > 42:
+            return 'Container Temperature above 42 deg C'
+
+
+class DriveInErrorDuringDataRun(FactIntervalCheck):
+    def inner_check(self):
+        if (
+                is_drive_error()
+                and is_data_taking()
+                and is_data_run()
+                ):
+            return 'Drive in Error during Data run'
+
+
+class BiasVoltageOnButNotCalibrated(FactIntervalCheck):
+    def inner_check(self):
+        is_voltage_on = sfc.status()['Bias_control'] == 'VoltageOn'
+        if (
+                is_voltage_on 
+                and is_feedback_not_calibrated() 
+                and sfc.voltages()['Med_voltage_in_V'] > 3
+                ):
+            return 'Bias voltage switched on, but bias crate not calibrated'
+
+class DIMNetworkNotAvailable(FactIntervalCheck):
+    def inner_check(self):
+        # can be checked this way according to:
+        # https://trac.fact-project.org/browser/trunk/FACT%2B%2B/src/smartfact.cc#L3131
+        if sfc.status()['DIM'] == 'Offline':
+            return 'DIM network not available'
+
+class NoDimCtrlServerAvailable(FactIntervalCheck):
+    def inner_check(self):
+        # Didn't find a clear way to check this, so I do:
+        if sfc.status()['Dim_Control'] in [
+                'Offline',
+                'NotReady',
+                'ERROR',
+                'FATAL',
+                ]:
+            return 'no dimctrl server available'
+
+
+class TriggerRateLowForTenMinutes(FactIntervalCheck):
+    history = []
+
+    def inner_check(self):
+        current_trigger_rate = sfc.trigger_rate()['Trigger_Rate_in_Bq']
+        self._append_to_history(current_trigger_rate)
+
+        df = pd.DataFrame(self.history)
+        if not df.empty and (df.rate < 1).all():
+            return 'Trigger rate < 1 Bq for 10 minutes'
+
+    def _append_to_history(self, rate):
+        self.history.append({'timestamp':datetime.utcnow(), 'rate':current_trigger_rate})
+        self._truncate_history()
```

Comment:
This is not good. It is really hidden here. An append method that removes stuff?